### DISCUSSION TITLE
How to use the ability to translate posts?

### DISCUSSION THREAD

**Discussion by TeeSee64** [mastodon/mastodon#20903] [Q&A]
Is there a how-to available on how to setup the ability to translate posts?

**Comment by veale** [comment]
If you are using DeepL
Sign up for the DeepL API free tier
edit your `.env.production` file

add the lines:
```
DEEPL_API_KEY=[your key]
DEEPL_PLAN=free
```

Restart Mastodon Web with 
`sudo systemctl restart mastodon-web`

**Reply by abdessalaam** [reply]
Thank you! Very helpful

**Comment by TeeSee64** [comment]
Thank you, I give it a try

**Comment by everton137** [comment]
@TeeSee64 By inspecting [Mastodon's code](https://github.com/mastodon/mastodon/blob/main/app/lib/translation_service.rb), I've discovered now this [LibreTranslate](https://libretranslate.com). I found these [instructions](https://sleeplessbeastie.eu/2022/11/17/how-to-use-libretranslate-with-mastodon/) on how to use it. I want to try later, since I cannot pay a DeepL plan at the moment.

**Reply by abdessalaam** [reply]
I would like LibreTranslate implementation as well... (please :))
Edit: particularly that DeepL doesn't support Arabic...

**Comment by veale** [comment]
It is implemented but you typically have to run a LibreTranslate instance, and this can take a lot of computational resources so is probably not to be done lightly.
